core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -332,6 +331,18 @@ public HandleCallbackDecorator getHandleCallbackDecorator() {
         return this.handleCallbackDecorator.get();
     }
 
+    /**
+     * Returns an internal handle supplier holder element used by this Jdbi instance.
+     * <br>
+     * <b>This is an internal method and not part of the public API!</b>
+     * @return A handle supplier holder element
+     * @param <T> type parameter
+     */
+    @Alpha
+    public <T extends HandleSupplierHolder> T getHandleSupplierHolder() {
+        return (T) handleSupplierHolder.updateAndGet(x -> x != null ? x : config.get(Handles.class).getHandleSupplierHolder());
```

Comment:
this is a little weird if you ever try to change the handle supplier holder, since this reference will never update.

Reply:
It allows changing the implementation class to be modified by calling 

```java
Jdbi jdbi = Jdbi.create(...);
jdbi.getConfig(Handles.class).setHandleSupplierHolderClass(MyMagicHandleSupplierHolder.class)
... more code
```

If the implementation would be created when the Jdbi is created, it could not be changed by configuration. If we allow changing at any time, there is a good chance that we might leak handles.

Open for better ideas, this is what I came up with.